### PR TITLE
[OV JS] Allow partial shape creation without parameter

### PR DESCRIPTION
**Pull request by vishniakov-nikolai**
### Details:
 - Call `getPartialShape` method on dynamic input/output lead to error. This change fix that.

### CODE REVIEW

File: src/bindings/js/node/src/partial_shape_wrap.cpp [hunk 1]
```diff
@@ -6,20 +6,27 @@
...
-    const size_t attrs_length = info.Length();
+    std::vector<std::string> allowed_signatures;
 
-    if (attrs_length == 1 && info[0].IsString()) {
-        try {
+    try {
+        if (ov::js::validate<Napi::String>(info, allowed_signatures)) {
             const auto& shape = std::string(info[0].ToString());
 
             _partial_shape = ov::PartialShape(shape);
```
Comment (almilosz) [inline src/bindings/js/node/src/partial_shape_wrap.cpp:18]:
```suggestion
            _partial_shape = ov::PartialShape(info[0].ToString());
```

File: src/bindings/js/node/src/partial_shape_wrap.cpp [hunk 2]
```diff
@@ -6,20 +6,27 @@
...
+            return;
+        } else if (ov::js::validate(info, allowed_signatures)) {
+            return;
         }
-    } else {
-        reportError(info.Env(), "Invalid parameters for PartialShape constructor.");
+
+        const auto error_message = ov::js::get_parameters_error_msg(info, allowed_signatures);
+
+        OPENVINO_THROW(error_message);
```
Comment (almilosz) [inline src/bindings/js/node/src/partial_shape_wrap.cpp:27]:
```suggestion
        OPENVINO_THROW("'PartialShape' constructor", ov::js::get_parameters_error_msg(info, allowed_signatures));
```

File: src/bindings/js/node/src/partial_shape_wrap.cpp [hunk 3]
```diff
@@ -6,20 +6,27 @@
...
-        try {
+    try {
+        if (ov::js::validate<Napi::String>(info, allowed_signatures)) {
             const auto& shape = std::string(info[0].ToString());
 
             _partial_shape = ov::PartialShape(shape);
-        } catch (std::exception& e) {
-            reportError(info.Env(), e.what());
+
+            return;
```
Comment (almilosz) [inline src/bindings/js/node/src/partial_shape_wrap.cpp:20]: This `return` can be removed if `OPENVINO_THROW` will be put in the `else` part

**Review by almilosz — APPROVED** [review]
In my opinion it's not a fix. You're just adding empty constructor for PartialShape ;)
Maybe a different title for the PR?

**Review by praasz — APPROVED** [review]

**Pull request merged**